addons/project_sale_expense/models/project_project.py

```diff
@@ -89,3 +89,19 @@ def get_action(res_ids):
             if expense_ids:
                 expense_data['costs']['action'] = get_action(expense_ids)
         return expense_data
+
+    def _get_already_included_profitability_invoice_line_ids(self):
+        move_line_ids = super()._get_already_included_profitability_invoice_line_ids()
+        query = self.env['hr.expense']._search([('is_refused', '=', False), ('state', 'in', ['approved', 'done'])])
+        query.add_where('hr_expense.analytic_distribution ? %s', [str(self.analytic_account_id.id)])
+        query.order = None
+        query_string, query_param = query.select('sale_order_id')
+        query_string = f"{query_string} GROUP BY sale_order_id"
+        self._cr.execute(query_string, query_param)
+        expenses_read_group = list(self._cr.dictfetchall())
+        if not expenses_read_group:
+            return move_line_ids
+        for res in expenses_read_group:
+            sale_order = self.env['sale.order'].browse(res['sale_order_id'])
+            move_line_ids.extend(sale_order.invoice_ids.mapped('invoice_line_ids').ids)
```

Comment:
If the SO has another product which is not really linked to expense in its configuration, we will exclude also this one, no?

Reply:
I think they are included before, when processing sale order lines, and they are listed under Other Services / Materials, so excluding them after won't be a problem (or maybe the good thing to do :sweat_smile: ) :

https://github.com/odoo/odoo/blob/e44d2fc7f882278e796113f37c4c7858297b83ff/addons/sale_project/models/project.py#L550-L554

![image](https://github.com/user-attachments/assets/fd677107-a8ca-4b7a-a236-21491973449f)
![image](https://github.com/user-attachments/assets/4fc27a5e-931d-40a6-9d48-628d1943cc8c)

Reply:
Hum, indeed. 🤔